Add index to campaign.name

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
Add index to campaign.name

Before
----------------------------------------
civicrm_campaign.name not indexed

After
----------------------------------------
Indexed on new installs

Technical Details
----------------------------------------

Comments
----------------------------------------
Note this related proposal https://lab.civicrm.org/dev/core/-/issues/2279

In this case I think the case for a index is fairly clear so that doesn't strictly apply, but it relates

mattwire commented:
@eileenmcnaughton How does this work on upgrades? Does System.updateindexes add this?

eileenmcnaughton commented:
@mattwire yes - if people call it. We stopped automatically adding invoices because we were trying to not have insanely slow upgrades too often and we added a system check. However, the index cleanup sometimes failed on old installs so we would up accepting that the check should be muted & we stopped working on fixing the routine at that point.

I think it would be good to create a page that helps to create missing indexes - probably as a core extension - which I mention in  https://lab.civicrm.org/dev/core/-/issues/2279 on a more granular basis. It's already possible to do only a single table with updateindexes.

I'm increasingly finding indexes are not 'one size fits all' as we have removed some core indexes (notably on contribution_status_id) to improve performance - but the idea is that at least on install the indexes should be generically sensible
